src/proto/semantifly.proto

```diff
@@ -0,0 +1,84 @@
+syntax = "proto3";
+
+package semantifly;
+
+option go_package = "accretional.com/semantifly/proto";
+
+service Semantifly {
+  rpc Add(AddRequest) returns (AddResponse) {}
+  rpc Delete(DeleteRequest) returns (DeleteResponse) {}
+  rpc Get(GetRequest) returns (GetResponse) {}
+  rpc Update(UpdateRequest) returns (UpdateResponse) {}
+  rpc LexicalSearch(LexicalSearchRequest) returns (LexicalSearchResponse) {}
+}
+
+message AddRequest {
+  string index_path = 1;
+  string data_type = 2;
+  string source_type = 3;
+  bool make_copy = 4;
+  repeated string data_uris = 5;
+}
+
+message FileContent {
+  string name = 1;
+  bytes content = 2;
+}
+
+message AddResponse {
+  bool success = 1;
+  string message = 2;
+}
+
+message DeleteRequest {
+  string index_path = 1;
+  bool delete_copy = 2;
+  repeated string data_uris = 3;
+}
+
+message DeleteResponse {
+  bool success = 1;
+  string message = 2;
+}
+
+message GetRequest {
+  string index_path = 1;
+  string name = 2;
+}
+
+message GetResponse {
+  bool success = 1;
+  string content = 2;
+  string message = 3;
+}
+
+message UpdateRequest {
+  string index_path = 1;
+  string name = 2;
+  string data_type = 3;
+  string source_type = 4;
+  bool update_copy = 5;
+  string data_uri = 6;
+}
+
+message UpdateResponse {
+  bool success = 1;
+  string message = 2;
+}
+
+message LexicalSearchRequest {
+  string index_path = 1;
+  string search_term = 2;
+  int32 top_n = 3;
+}
+
+message LexicalSearchResponse {
+  bool success = 1;
```

Comment:
Don't use success fields in proto responses, grpc has builtin functions to check that

Reply:
fixed